rsc/src/main/java/com/cloudera/livy/rsc/driver/RSCDriver.java

```diff
@@ -393,6 +396,22 @@ public void handle(ChannelHandlerContext ctx, BypassJobRequest msg) throws Excep
     }
   }
 
+  public BypassJobWrapper initiateWrapperBasedOnKind(BypassJobRequest msg) throws Exception {
```

Comment:
Why is this method here?

Instead, I think it would be easier to have a method:

```
protected BypassJobWrapper createWrapper(BypassJobRequest msg) throws Exception 
```

The default implementation creates the Java wrapper. In `ReplDriver` you can override this method, and in the case of a python session, you can delegate it to a new method in the `PythonInterpreter` instance. Then there's no need for python-specific anything in the `rsc/` module. And you don't need to keep anything in static variables as you were asking about on chat.